### PR TITLE
Change **chekpoint to **kwargs

### PR DESCRIPTION
**Pull request by MrNightSky**
Change extra arguments into pack_checkpoint function to save all parameters. Need for resume option.

### Pull Request FAQ
- [documentation](https://catalyst-team.github.io/catalyst/)
- [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### Checklist
- [ ] Have you updated tests for the new functionality?
- [ ] Have you added your new classes/functions to the docs?
- [ ] Have you updated the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Have you run [colab minimal CI/CD](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/colab_ci_cd.ipynb) with `latest` and `minimal` requirements?
- [ ] Have you checked XLA integration with [single](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_single_process.ipynb) and [multiple](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_multi_process.ipynb) processes?

### CODE REVIEW

**Comment by MrNightSky** [comment]
Can add the same for ApexEngine after approval.

File: catalyst/engines/amp.py [hunk 1]
```diff
@@ -100,13 +100,13 @@ def pack_checkpoint(
             ``criterion_state_dict``, ``optimizer_state_dict``,
             ``scheduler_state_dict`` keys.
         """
-        checkpoint = {"scaler": self.scaler.state_dict()}
+        kwargs["scaler"] = self.scaler.state_dict()
```
Comment (bagxi) [inline catalyst/engines/amp.py:103]: What is the point of these changes?

File: catalyst/engines/amp.py [hunk 2]
```diff
@@ -386,13 +386,13 @@ def pack_checkpoint(
...
```
Comment (bagxi) [inline catalyst/engines/amp.py:389]: What is the point of these changes?

**Review by bagxi — DISMISSED** [review]

File: catalyst/engines/amp.py (hunk 2 above)
Comment (MrNightSky) [reply catalyst/engines/amp.py:389]: We need to transfer kwargs to save extra features like "global_epoch_step" and other. Without them resume option doesn't work.

File: catalyst/engines/amp.py (hunk 1 above)
Comment (MrNightSky) [reply catalyst/engines/amp.py:103]: We need to transfer kwargs to save extra features like "global_epoch_step" and other. Without them resume option doesn't work.

Comment (bagxi) [reply catalyst/engines/amp.py:103]: Why not to do
```
checkpoint = super().pack_checkpoint(
    model=model,
     criterion=criterion,
     optimizer=optimizer,
     scheduler=scheduler,
      **checkpoint,
      **kwargs,
)
```
than?

Comment (MrNightSky) [reply catalyst/engines/amp.py:103]: The same
You can do this variant, not so important for me.

**Comment by Scitator** [comment]
@MrNightSky could you please also fix `apex` version?

**Comment by MrNightSky** [comment]
@Scitator sorry, that I didn't answer so long. Too much work before New Year

**Pull request merged**